### PR TITLE
added new module lofraq/viterbi; solves new module: lofreq/viterbi #5158

### LINKED ISSUE #5158: new module: lofreq/viterbi
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by MarieLataretu**
## PR checklist

Closes #5158 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by MarieLataretu** [comment]
Why is it running so many lofreq-unrelated tests? 😱

**Comment by FloWuenne** [comment]
> Why is it running so many lofreq-unrelated tests? 😱

The reason why it runs unrelated tests is that it compares your PR changes to master and so any others changes done while you made your feature branch will be run as tests as well. Usually, clicking Update branch at the bottom of the PR should bring it to only running your tests!

**Comment by MarieLataretu** [comment]
okay, the problem with the failing conda test:

The `samtools` version differs in the conda (1.19.2) environment and the container (1.17). This leads to a different header line in the BAM file, even if the build version is the same.

**Comment by FloWuenne** [comment]
@MarieLataretu Great catch! I was also looking at the difference in conda vs container environment. This is really strange, as the biocontainers build from bioconda should be exactly the same no? Would this be fixed by using a different build? 
Otherwise I see two options:
1) Disable conda
2) Generate a custom container image with the samtools version corresponding to the conda environment

File: modules/nf-core/lofreq/viterbi/environment.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
```
Comment (FloWuenne) [inline modules/nf-core/lofreq/viterbi/environment.yml:2]:
```suggestion
```

**Comment by MarieLataretu** [comment]
I checked to which build the conda env was resolved, and adapted the singularity container accordingly (so far only locally - I don't know how conda decides on the build 🤔 )

My best guess is that in conda, samtools always resolves to the latest version when you build the environment. The same is true for the container, only that it's frozen to the latest samtools version when the container is built.

I checked out a third option, but I don't know if follows the nf-core standards: I added the samtools version from the container as a dependency to the yml:
```
dependencies:
  - bioconda::lofreq=2.1.5
  - bioconda::samtools=1.17
 ```

Locally the tests pass!

File: modules/nf-core/lofreq/viterbi/environment.yml (hunk 1 above)
Comment (FloWuenne) [reply modules/nf-core/lofreq/viterbi/environment.yml:2]: I would remove these comments from the environment.yml for a clean yml file.

**Comment by FloWuenne** [comment]
> I checked to which build the conda env was resolved, and adapted the singularity container accordingly (so far only locally - I don't know how conda decides on the build 🤔 )
> 
> My best guess is that in conda, samtools always resolves to the latest version when you build the environment. The same is true for the container, only that it's frozen to the latest samtools version when the container is built.
> 
> I checked out a third option, but I don't know if follows the nf-core standards: I added the samtools version from the container as a dependency to the yml:
> 
> ```
> dependencies:
>   - bioconda::lofreq=2.1.5
>   - bioconda::samtools=1.17
> ```
> 
> Locally the tests pass!

I just tested your patch with specifying the samtools version in the `environment.yml` file and nf-test still produces different md5sums with conda than the ones produced by docker and singularity, which suggests to me also potential other differences than just the samtool version?

**Review by FloWuenne — CHANGES_REQUESTED** [review]
I spend some time in Gitpod trying to modify the conda dependencies so that they would output the same md5sum tests as docker and singularity, but was also unable to do so. For the PR to pass, this is the main thing to resolve...

**Comment by MarieLataretu** [comment]
(I'm struggling with git 😅 )

**Review by FloWuenne — APPROVED** [review]
LGTM ⭐️

**Pull request merged**